commit f8d9342e8fe1d27ebdf304b283d2f4ec77b77e02 (refs/changes/16/4716/2)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-08-17T20:18:19-04:00 (1 year, 2 months ago)
    
    joyent/node-manta#327 completion tests could ignore hidden files
    Reviewed by: Jordan Hendricks <jordan.hendricks@joyent.com>
    Approved by: Jordan Hendricks <jordan.hendricks@joyent.com>

diff --git a/test/completion.test.js b/test/completion.test.js
index 09a8745..9b4035e 100644
--- a/test/completion.test.js
+++ b/test/completion.test.js
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -7,17 +7,17 @@
  * that looks somewhat like Bash completion code.
  */
 
-var exec = require('child_process').exec;
 var fs = require('fs');
 var path = require('path');
 
+var forkExecWait = require('forkexec').forkExecWait;
+
 /*
  * Globals
  */
 
 var binDir = path.resolve(__dirname, '..', 'bin');
 
-
 /*
  * Helper functions
  */
@@ -26,20 +26,25 @@ function test(name, testfunc) {
     module.exports[name] = testfunc;
 }
 
-
 /*
  * Tests
  */
 
 fs.readdirSync(binDir).forEach(function (name) {
-    var cmd = path.join(binDir, name);
+    // node-manta#327 completion tests could ignore hidden files
+    if (name[0] === '.') {
+        return;
+    }
 
+    var cmd = path.join(binDir, name);
     test(name + ' --completion', function (t) {
-        exec(cmd + ' --completion', function (err, stdout, stderr) {
+        forkExecWait({
+            argv: [cmd, '--completion']
+        }, function (err, info) {
             t.ifError(err);
-            t.equal(stderr, '',
+            t.equal(info.stderr, '',
                 'no stderr output from "' + name + ' --completion"');
-            t.ok(/COMPREPLY/.test(stdout), 'stdout from "' + name +
+            t.ok(/COMPREPLY/.test(info.stdout), 'stdout from "' + name +
                 ' --completion" looks like Bash completion code');
             t.done();
         });
